src/Compilers/Core/Portable/SourceGeneration/GeneratorContexts.cs

```diff
@@ -148,6 +168,39 @@ private static void CheckIsEmpty<T>(T x)
         }
     }
 
+    /// <summary>
+    /// Context passed to a source generator when it has opted-in to PostInitialization via <see cref="GeneratorInitializationContext.RegisterForPostInitialization(Action{GeneratorPostInitializationContext})"/>
+    /// </summary>
+    public readonly struct GeneratorPostInitializationContext
+    {
+        private readonly AdditionalSourcesCollection _additionalSources;
+
+        internal GeneratorPostInitializationContext(AdditionalSourcesCollection additionalSources, CancellationToken cancellationToken)
+        {
+            _additionalSources = additionalSources;
+            CancellationToken = cancellationToken;
+        }
+
+        /// <summary>
+        /// A <see cref="CancellationToken"/> that can be checked to see if the PostInitialization should be cancelled.
+        /// </summary>
+        public CancellationToken CancellationToken { get; }
+
+        /// <summary>
+        /// Adds source code in the form of a <see cref="string"/> to the compilation that will be availble during subsequent phases
+        /// </summary>
+        /// <param name="hintName">An identifier that can be used to reference this source text, must be unique within this generator</param>
+        /// <param name="source">The source code to be add to the compilation</param>
+        public void AddSource(string hintName, string source) => AddSource(hintName, SourceText.From(source, Encoding.UTF8));
```

Comment:
Is this overload necessary, or could we simply include the overload below?

Reply:
We don't require it, but it matches the execution context, and was a common ask there, so I think it's worth having them match in this context.

---
In reply to: [564876015](https://github.com/dotnet/roslyn/pull/50026#discussion_r564876015) [](ancestors = 564876015)